From 971d6f7a47cc4842af304f0e22fe0e8ca574c3e3 Mon Sep 17 00:00:00 2001
From: Wallace Wang <r59996@freescale.com>
Date: Sat, 27 Sep 2008 16:28:17 +0800
Subject: [PATCH] ENGR00093249 MX51: Remove the GPIO repetitive request

SPDIF request the GPIO EIM_D16
AUDIO request the GPIO EIM_D16
Resolve the compile warning

Signed-off-by: Wallace Wang <r59996@freescale.com>
---
 arch/arm/mach-mx51/mx51_3stack_gpio.c |   13 ++-----------
 1 files changed, 2 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-mx51/mx51_3stack_gpio.c b/arch/arm/mach-mx51/mx51_3stack_gpio.c
index 09bf21a..87fa180 100644
--- a/arch/arm/mach-mx51/mx51_3stack_gpio.c
+++ b/arch/arm/mach-mx51/mx51_3stack_gpio.c
@@ -457,7 +457,7 @@ void gpio_i2c_inactive(int i2c_num)
 
 EXPORT_SYMBOL(gpio_i2c_inactive);
 
-void gpio_i2c_hs_active()
+void gpio_i2c_hs_active(void)
 {
 	mxc_request_iomux(MX51_PIN_I2C1_CLK,
 			  IOMUX_CONFIG_ALT0 | IOMUX_CONFIG_SION);
@@ -470,7 +470,7 @@ void gpio_i2c_hs_active()
 
 EXPORT_SYMBOL(gpio_i2c_hs_active);
 
-void gpio_i2c_hs_inactive()
+void gpio_i2c_hs_inactive(void)
 {
 }
 
@@ -518,15 +518,6 @@ void gpio_activate_audio_ports(void)
 	mxc_request_iomux(MX51_PIN_AUD3_BB_FS, IOMUX_CONFIG_ALT0);
 	mxc_iomux_set_pad(MX51_PIN_AUD3_BB_FS, pad_val);
 
-	/*
-	 * CPU3 AAPL board:
-	 * codec mclk is driven by 12MHz external xtal oscillator.
-	 * enable it by driving EIM_D16 high.
-	 */
-	mxc_request_iomux(MX51_PIN_EIM_D16, IOMUX_CONFIG_ALT1);
-	mxc_iomux_set_pad(MX51_PIN_EIM_D16, pad_val);
-	mxc_set_gpio_direction(MX51_PIN_EIM_D16, 0);
-	mxc_set_gpio_dataout(MX51_PIN_EIM_D16, 1);
 }
 
 EXPORT_SYMBOL(gpio_activate_audio_ports);
-- 
1.5.4.4

